evap/staff/views.py

```diff
@@ -1386,23 +1388,67 @@ def evaluation_email(request, evaluation_id):
     )
 
 
-def helper_delete_users_from_evaluation(evaluation, operation):
-    if "participants" in operation:
-        deleted_person_count = evaluation.participants.count()
-        deletion_message = _("{} participants were deleted from evaluation {}")
-        evaluation.participants.clear()
+class ImportAction(Enum):
+    COPY = "copy"
+    IMPORT = "import"
+    TEST = "test"
+
+    @classmethod
+    def from_operation(cls, operation: str) -> "ImportAction":
+        if operation.startswith("copy-"):
+            return cls.COPY
+        if operation.startswith("import-"):
+            return cls.IMPORT
+        if operation.startswith("test-"):
+            return cls.TEST
+        raise ValueError(f"Unknown operation: {operation}")
+
+
+def import_or_copy_participants(
+    request,
+    replace: bool,
+    import_action: Literal[ImportAction.COPY, ImportAction.IMPORT],
+    import_type: ImportType,
+    evaluation: Evaluation,
+    copy_form: EvaluationParticipantCopyForm,
+) -> bool:
+    if replace:
+        if import_type == ImportType.PARTICIPANT:
+            deleted_person_count = evaluation.participants.count()
+            deletion_message = _("{} participants were deleted from evaluation {}")
+            evaluation.participants.clear()
+        elif import_type == ImportType.CONTRIBUTOR:
+            deleted_person_count = evaluation.contributions.exclude(contributor=None).count()
+            deletion_message = _("{} contributors were deleted from evaluation {}")
+            evaluation.contributions.exclude(contributor=None).delete()
```

Comment:
Maybe add `else: assert_never(import_type)`?

Reply:
done, but it needs a type: ignore instead, which is fixed on master though